**ASSIGNMENT NO: 4 Statistics & Probability**

In [ ]:
import pandas as pd

# Load dataset
df = pd.read_csv('telecom_customer_churn_cleaned.csv')

# Basic statistics
mean_values = df.mean(numeric_only=True)
median_values = df.median(numeric_only=True)
mode_values = df.mode(numeric_only=True).iloc[0]
variance_values = df.var(numeric_only=True)

print("Mean:\n", mean_values, "\n")
print("Median:\n", median_values, "\n")
print("Mode:\n", mode_values, "\n")
print("Variance:\n", variance_values, "\n")

# Correlation matrix
correlation = df.corr(numeric_only=True)
print("Correlation Matrix:\n", correlation)


Mean:
 Age                                     46.488661
Number of Dependents                     0.468583
Zip Code                             93492.072512
Latitude                                36.203815
Longitude                             -119.761209
Number of Referrals                      1.950744
Tenure in Months                        32.432183
Avg Monthly Long Distance Charges       25.416235
Avg Monthly GB Download                 26.192882
Monthly Charge                          64.792366
Total Charges                         2284.741079
Total Refunds                            1.941131
Total Extra Data Charges                 6.871299
Total Long Distance Charges            750.030946
Total Revenue                         3039.702193
dtype: float64 

Median:
 Age                                     46.000000
Number of Dependents                     0.000000
Zip Code                             93522.000000
Latitude                                36.220761
Longitude        

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


Correlation analysis. Identifying the 3 features most related to target variable

In [ ]:
# Convert target to numeric (Churned=1, Stayed=0)
df['Customer Status'] = df['Customer Status'].map({'Churned': 1, 'Stayed': 0})

# Correlation with target
corr_with_target = df.corr(numeric_only=True)['Customer Status'].sort_values(ascending=False)
print("Correlation with Target Variable:\n", corr_with_target)

# Top 3 features most correlated
top3 = corr_with_target[1:4]
print("\nTop 3 Features Most Related to Churn:\n", top3)


Correlation with Target Variable:
 Customer Status                      1.000000
Monthly Charge                       0.175817
Age                                  0.112251
Longitude                            0.024238
Total Extra Data Charges            -0.000525
Avg Monthly Long Distance Charges   -0.000968
Zip Code                            -0.018547
Total Refunds                       -0.043052
Latitude                            -0.043680
Avg Monthly GB Download             -0.095124
Number of Dependents                -0.232172
Total Charges                       -0.248040
Total Long Distance Charges         -0.267329
Total Revenue                       -0.276753
Number of Referrals                 -0.312673
Tenure in Months                    -0.432305
Name: Customer Status, dtype: float64

Top 3 Features Most Related to Churn:
 Monthly Charge    0.175817
Age               0.112251
Longitude         0.024238
Name: Customer Status, dtype: float64


Top 3 Correlated Features with Customer Churn:

Monthly Charge     0.175817

Age                0.112251

Longitude          0.024238

Name: Customer Status, dtype: float64